### PR TITLE
Show note

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This shows a note, if available, in file detail sharing fragment.
As it is only shown if a note is available, it is out of the box backward compatible.

Requires PR for server: https://github.com/nextcloud/server/pull/12978

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky corresponding lib change has been merged to master branch and jitpackk build has been triggered and turned green, so jitpacak lib master will ship the lib change now 👍

**Comment by AndyScherzinger** [comment]
pinging @schiessle for feedback/input ❤️

**Comment by tobiasKaminsky** [comment]
image

Is the note "ShareAble" good enough to recognize? (both on Android / web UI)

**Comment by AndyScherzinger** [comment]
> Is the note "ShareAble" good enough to recognize? (both on Android / web UI)

I don't think so to be honest. It could maybe have the text file icon or something. Maybe @nextcloud/designers have a nice idea?

**Comment by jancborchardt** [comment]
> > Is the note "ShareAble" good enough to recognize? (both on Android / web UI)
> 
> I don't think so to be honest. It could maybe have the text file icon or something. Maybe @nextcloud/designers have a nice idea?

I’d say it’s good for the first pass – normally the note has some more explanatory content than just a word, and it’s inherently understandable as a note.

If we see it’s a problem in actual use, we can always add `.icon-edit`, the text document icon we also use for the "Add note" menu entry.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky needs a rebase and one question: is this ready to merge or does it need the server PR to be merged?

**Comment by tobiasKaminsky** [comment]
> @tobiasKaminsky needs a rebase and one question: is this ready to merge or does it need the server PR to be merged?

This originally depended on #3320, but I rebased it onto master, which now results in duplicated classes/additions, so I would like to merge #3320 before this.
After merge of #3320 I'll do a rebase.

This still requires https://github.com/nextcloud/server/pull/12978 to be merged.
Backward compability should be working out of the box as older server just do not respond to "nc:note".
But I'll test it then.

(sorry for the confusion)

File: src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 1]
```diff
@@ -872,6 +873,16 @@ Drawable doAvatarInBackground() {
 
             int px = getAvatarDimension();
 
+            boolean notOnSameServer = false;
+
+            if (notOnSameServer) {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: @tobiasKaminsky this is always false...

File: src/main/res/layout/file_details_sharing_fragment.xml [hunk 2]
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (AndyScherzinger) [inline src/main/res/layout/file_details_sharing_fragment.xml]: linebreak before the comment :)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky just two minor comments plus needs-rebase :)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky see review comments :)

File: src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: Good question, why I did this.
Maybe to test…
I'll fix it.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky rebase 'n' merge? ❤️

**Comment by tobiasKaminsky** [comment]
Not quite…the part with federated icons is still open.
I hope I find time tomorrow :/

File: src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: We ignore this for now. If there is a federated share, only the text generated avatar will be shown.
Ref: https://github.com/nextcloud/server/issues/14564

**Comment by tobiasKaminsky** [comment]
Finally done :-)

**Pull request merged**